### PR TITLE
platform: allow primary core stack size config.

### PR DESCRIPTION
**Pull request by lkoenig**
Allow to set the primary core stack size using the Kconfig mechanism.

### CODE REVIEW

File: src/platform/Kconfig [hunk 1]
```diff
@@ -530,4 +530,10 @@ config FORCE_DMA_COPY_WHOLE_BLOCK
 	  The host component forces DMA to copy the block size to avoid
 	  copying byte jitter between the components of the same pipeline.
 
+config SOF_STACK_SIZE
+	int "Primary core SOF stack size"
+	default 4096
+	help
+	  Size of the primary core stack.
```
Comment (lgirdwood) [inline src/platform/Kconfig:537]: Can we have a description of why would would want to change this. Fwiw, I think this is already an option with Zephyr.

**Comment by lkoenig** [comment]
This change is required to allow more complex algorithm to be instantiated at create.
On Tigerlake, Google AEC requires a deeper stack size at create. 
Making it configurable via Kconfig allow platform requiring deeper stack to have those while keeping the current stack definition on all others.
I am really looking forward Zephyr. However my understanding is that it is not yet ready for Tigerlake and Alder lake which we are targeting for the AEC.

**Comment by lgirdwood** [comment]
> This change is required to allow more complex algorithm to be instantiated at create. On Tigerlake, Google AEC requires a deeper stack size at create. Making it configurable via Kconfig allow platform requiring deeper stack to have those while keeping the current stack definition on all others. I am really looking forward Zephyr. However my understanding is that it is not yet ready for Tigerlake and Alder lake which we are targeting for the AEC.

Sorry, I should have clarified to add the description in the Kconfig help. i.e. to help out other developers understand the option and where to apply it.

**Comment by lkoenig** [comment]
Sorry for the misunderstanding.
@lgirdwood let me know is the description I put in the `Kconfig` is appropriate.

**Review by lgirdwood — APPROVED** [review]

**Review by dbaluta — APPROVED** [review]

**Pull request merged**